### PR TITLE
added: support for mixed multipatch adaptive simulations

### PR DESCRIPTION
**Pull request by akva2**
Extracted from https://github.com/OPM/IFEM/pull/255

I left out the dd part since it's not yet working and not yet required.

It's time to get this in.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-stokes=280 please

File: src/ASM/LR/GlobalNodes.C [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+  for (const auto& pch : pchs) {
+    it->resize(pch->nBasisFunctions());
+    std::iota(it->begin(), it->end(), nNodes);
+    nNodes += pch->nBasisFunctions();
+    ++it;
+  }
+
+  // remap common nodes
+  InterfaceOrder ifOrder;
+  std::set<ASM::Interface, InterfaceOrder> ifset(ifOrder);
```
Comment (kmokstad) [inline src/ASM/LR/GlobalNodes.C:158]: Is it necessary with the `ifOrder` instance here? Thought the following would work fine:

    std::set<ASM::Interface,InterfaceOrder> ifset;

We have a similar construct for the `MPC` sets.
Even better, could the `InterfaceOrder` class be replaced by a local lambda here?

File: src/SIM/SIMinput.C [hunk 2]
```diff
@@ -1604,9 +1604,16 @@ bool SIMinput::refine (const LR::RefineData& prm, Vectors& sol)
...
-        if ((locId = myModel[i]->getNodeIndex(k+1)) > 0)
-          if (refineIndices[i].insert(locId-1).second)
+        if (!prm.MLGN.empty()) {
+          const auto it = std::find(prm.MLGN[i].begin(), prm.MLGN[i].end(), k);
+          if (it != prm.MLGN[i].end())
+            if (refineIndices[i].insert(std::distance(prm.MLGN[i].begin(), it)).second)
+              changed = true;
+        } else {
+         if ((locId = myModel[i]->getNodeIndex(k+1)) > 0)
+           if (refineIndices[i].insert(locId-1).second)
```
Comment (kmokstad) [inline src/SIM/SIMinput.C:1614]: check indents (probably some tabs' involved)

File: src/ASM/LR/GlobalNodes.C (hunk 1 above)
Comment (akva2) [reply src/ASM/LR/GlobalNodes.C]: removed the instance. cannot use a lambda as a tempalte parameter.

File: src/SIM/SIMinput.C (hunk 2 above)
Comment (akva2) [reply src/SIM/SIMinput.C]: fixed

**Comment by akva2** [comment]
jenkins build this with downstreams ifem-stokes=280 please

**Pull request merged**